File: Homeworks/C#2/Methods/ReverseNumber/ReverseNumber.cs
//Write a method that reverses the digits of given decimal number.

namespace ReverseNumber
{
    using System;
    using System.Text;

    class ReverseNumber
    {
        static void Main(string[] args)
        {
            Console.Write("Enter a number: ");
            string input = Console.ReadLine();

            Console.WriteLine(Reverse(input));
        }

        private static double Reverse(string number)
        {
            char[] num = number.ToCharArray();
            char[] result = new char[num.Length];

            for (int index = 0; index < result.Length; index++) 
                if (!Char.IsDigit(num[index])) 
                    result[index] = num[index];

            for (int resultInd = result.Length - 1, numInd = 0;
                 resultInd >= 0 && numInd < num.Length;
                 resultInd--, numInd++) 
            {
                while (result[resultInd]!='\0')
                    --resultInd;

                while (!Char.IsDigit(num[numInd])) 
                    ++numInd;

                result[resultInd] = num[numInd];
            }

            return double.Parse(new String(result));
        }
    }
}

File: Homeworks/JS-basics/Functions/reverseNumber.js
// Write a function that reverses the digits of given
// decimal number. Example: 256 ? 652

console.log(reverseNumber(256));
console.log(reverseNumber(0));
console.log(reverseNumber(12345));

function reverseNumber(number){
    var result = 0;

    while(number){
        result = result * 10 + number % 10;
        number = Math.floor(number / 10);
    }

    return result;
}